aten/src/ATen/native/Copy.cpp

```diff
@@ -296,8 +296,8 @@ static Tensor & copy_impl(Tensor & self, const Tensor & src, bool non_blocking)
   }
 #endif
 
-  if(!self.is_complex() && src.is_complex()) {
-    TORCH_WARN_ONCE("Casting complex values to real discards the imaginary part");
+if(!self.is_complex() && src.is_complex()) {
+    TORCH_WARN_ONCE("Casting complex values to real discards the imaginary part for non-bool casts");
```

Comment:
Just throw this warning if `self` is not bool

Reply:
Great point 😅